### PR TITLE
Don't allow reason in vote request if shutter enabled

### LINKED ISSUE #491: Don't allow reason on shutter proposals
Related to https://github.com/snapshot-labs/snapshot/issues/3228

### PR DESCRIPTION
**Pull request by emnul**
Fixes #491 .

Changes proposed in this pull request:
- Reject requests that have a reason when shutter is enabled

### CODE REVIEW

File: src/ingestor/writer/vote.ts [hunk 1]
```diff
@@ -39,6 +39,8 @@ export async function verify(body): Promise<any> {
   if (proposal.privacy === 'shutter') {
     if (typeof msg.payload.choice !== 'string' || !msg.payload.choice.startsWith('0x'))
       return Promise.reject('invalid choice');
+    if (msg.payload.reason.length !== 0)
+      return Promise.reject('no reason allowed on proposal with Shutter privacy enabled');
```
Comment (ChaituVR) [inline src/ingestor/writer/vote.ts:43]:
```suggestion
    if (msg.payload.reason?.length)
      return Promise.reject('reason not allowed when shutter is enabled');
```

**Review by ChaituVR — APPROVED** [review]
utACK

**Comment by samuveth** [comment]
Any reason this still isn't merged?

**Comment by ChaituVR** [comment]
@emnul Can resolve conflicts?

**Comment by samuveth** [comment]
@ChaituVR, could you please take care of this? It seems like @emnul is unresponsive.

**Comment by bonustrack** [comment]
Closing this issue, this is not relevant anymore.

**Pull request closed without merging**